app/Models/PostingPeriod.php

```
@@ -55,4 +55,23 @@ public function scopeWithDetails($query)
+            if ($currentMonth) {
+                $subPeriodsQuery->where('start_date', '!=', $currentMonth->startOfMonth());
+            }
+    
+            if ($subPeriodsQuery->where('status', PostingPeriodStatusType::OPEN->value)->exists()) {
+                return true;
+            } else {
+                $query->update([
+                    'status' => PostingPeriodStatusType::CLOSED,
+                ]);
+                return false;
+            }
```

PHPMD (2.15.0)

The method scopeCheckIfStatusIsOpenYearly uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (Clean Code Rules)

(ElseExpression)